diff -Naur gettext-0.18.1.1.orig/gettext-runtime/intl/localename.c gettext-0.18.1.1/gettext-runtime/intl/localename.c
--- gettext-0.18.1.1.orig/gettext-runtime/intl/localename.c	2012-01-23 13:09:27.778200034 -0500
+++ gettext-0.18.1.1/gettext-runtime/intl/localename.c	2012-01-23 13:12:12.698979684 -0500
@@ -2607,7 +2607,7 @@
     locale_t thread_locale = uselocale (NULL);
     if (thread_locale != LC_GLOBAL_LOCALE)
       {
-#  if __GLIBC__ >= 2
+#  if __GLIBC__ >= 2 && !defined(__UCLIBC__)
         /* Work around an incorrect definition of the _NL_LOCALE_NAME macro in
            glibc < 2.12.
            See <http://sourceware.org/bugzilla/show_bug.cgi?id=10968>.  */
@@ -2771,7 +2771,7 @@
    However it does not specify the exact format.  Neither do SUSV2 and
    ISO C 99.  So we can use this feature only on selected systems (e.g.
    those using GNU C Library).  */
-#if defined _LIBC || (defined __GLIBC__ && __GLIBC__ >= 2)
+#if defined _LIBC || (defined __GLIBC__ && __GLIBC__ >= 2 && (!defined __UCLIBC__ || defined __UCLIBC_HAS_LOCALE__))
 # define HAVE_LOCALE_NULL
 #endif
 
diff -Naur gettext-0.18.1.1.orig/gettext-tools/gnulib-lib/localename.c gettext-0.18.1.1/gettext-tools/gnulib-lib/localename.c
--- gettext-0.18.1.1.orig/gettext-tools/gnulib-lib/localename.c	2012-01-23 13:09:27.889450470 -0500
+++ gettext-0.18.1.1/gettext-tools/gnulib-lib/localename.c	2012-01-23 13:12:19.669375208 -0500
@@ -2607,7 +2607,7 @@
     locale_t thread_locale = uselocale (NULL);
     if (thread_locale != LC_GLOBAL_LOCALE)
       {
-#  if __GLIBC__ >= 2
+#  if __GLIBC__ >= 2 && !defined(__UCLIBC__)
         /* Work around an incorrect definition of the _NL_LOCALE_NAME macro in
            glibc < 2.12.
            See <http://sourceware.org/bugzilla/show_bug.cgi?id=10968>.  */
@@ -2771,7 +2771,7 @@
    However it does not specify the exact format.  Neither do SUSV2 and
    ISO C 99.  So we can use this feature only on selected systems (e.g.
    those using GNU C Library).  */
-#if defined _LIBC || (defined __GLIBC__ && __GLIBC__ >= 2)
+#if defined _LIBC || (defined __GLIBC__ && __GLIBC__ >= 2 && (!defined __UCLIBC__ || defined __UCLIBC_HAS_LOCALE__))
 # define HAVE_LOCALE_NULL
 #endif
 
