diff -up gettext-0.22.3/gettext-tools/examples/Makefile.am.1~ gettext-0.22.3/gettext-tools/examples/Makefile.am
--- gettext-0.22.3/gettext-tools/examples/Makefile.am.1~	2023-09-18 22:09:26.000000000 +0200
+++ gettext-0.22.3/gettext-tools/examples/Makefile.am	2023-10-05 15:43:19.133949408 +0200
@@ -98,43 +98,6 @@ EXAMPLESFILES = \
   hello-c++-qt/m4/qt.m4 \
   hello-c++-qt/po/Makefile.am \
   \
-  hello-c++-kde/INSTALL \
-  hello-c++-kde/BUGS \
-  hello-c++-kde/autogen.sh \
-  hello-c++-kde/autoclean.sh \
-  hello-c++-kde/hello.cc \
-  hello-c++-kde/hellowindow.h \
-  hello-c++-kde/hellowindow.cc \
-  hello-c++-kde/Makefile.am \
-  hello-c++-kde/configure.in.in \
-  hello-c++-kde/m4/Makefile.am \
-  hello-c++-kde/admin/acinclude.m4.in \
-  hello-c++-kde/admin/am_edit \
-  hello-c++-kde/admin/ChangeLog \
-  hello-c++-kde/admin/compile \
-  hello-c++-kde/admin/conf.change.pl \
-  hello-c++-kde/admin/config.guess \
-  hello-c++-kde/admin/config.pl \
-  hello-c++-kde/admin/config.sub \
-  hello-c++-kde/admin/configure.in.bot.end \
-  hello-c++-kde/admin/configure.in.min \
-  hello-c++-kde/admin/cvs-clean.pl \
-  hello-c++-kde/admin/cvs.sh \
-  hello-c++-kde/admin/debianrules \
-  hello-c++-kde/admin/depcomp \
-  hello-c++-kde/admin/detect-autoconf.sh \
-  hello-c++-kde/admin/Doxyfile.am \
-  hello-c++-kde/admin/Doxyfile.global \
-  hello-c++-kde/admin/install-sh \
-  hello-c++-kde/admin/libtool.m4.in \
-  hello-c++-kde/admin/ltmain.sh \
-  hello-c++-kde/admin/Makefile.common \
-  hello-c++-kde/admin/missing \
-  hello-c++-kde/admin/mkinstalldirs \
-  hello-c++-kde/admin/ylwrap \
-  hello-c++-kde/po/Makevars \
-  hello-c++-kde/po/POTFILES.in \
-  \
   hello-c++-gnome/INSTALL \
   hello-c++-gnome/autogen.sh \
   hello-c++-gnome/autoclean.sh \
@@ -389,7 +352,6 @@ EXAMPLESDIRS = \
   hello-c++ \
   hello-c++20 \
   hello-c++-qt \
-  hello-c++-kde \
   hello-c++-gnome \
   hello-c++-wxwidgets \
   hello-objc \
diff -up gettext-0.22.3/gettext-tools/examples/Makefile.in.1~ gettext-0.22.3/gettext-tools/examples/Makefile.in
--- gettext-0.22.3/gettext-tools/examples/Makefile.in.1~	2023-10-04 16:56:55.000000000 +0200
+++ gettext-0.22.3/gettext-tools/examples/Makefile.in	2023-10-05 15:43:26.461028914 +0200
@@ -408,43 +408,6 @@ EXAMPLESFILES = \
   hello-c++-qt/m4/qt.m4 \
   hello-c++-qt/po/Makefile.am \
   \
-  hello-c++-kde/INSTALL \
-  hello-c++-kde/BUGS \
-  hello-c++-kde/autogen.sh \
-  hello-c++-kde/autoclean.sh \
-  hello-c++-kde/hello.cc \
-  hello-c++-kde/hellowindow.h \
-  hello-c++-kde/hellowindow.cc \
-  hello-c++-kde/Makefile.am \
-  hello-c++-kde/configure.in.in \
-  hello-c++-kde/m4/Makefile.am \
-  hello-c++-kde/admin/acinclude.m4.in \
-  hello-c++-kde/admin/am_edit \
-  hello-c++-kde/admin/ChangeLog \
-  hello-c++-kde/admin/compile \
-  hello-c++-kde/admin/conf.change.pl \
-  hello-c++-kde/admin/config.guess \
-  hello-c++-kde/admin/config.pl \
-  hello-c++-kde/admin/config.sub \
-  hello-c++-kde/admin/configure.in.bot.end \
-  hello-c++-kde/admin/configure.in.min \
-  hello-c++-kde/admin/cvs-clean.pl \
-  hello-c++-kde/admin/cvs.sh \
-  hello-c++-kde/admin/debianrules \
-  hello-c++-kde/admin/depcomp \
-  hello-c++-kde/admin/detect-autoconf.sh \
-  hello-c++-kde/admin/Doxyfile.am \
-  hello-c++-kde/admin/Doxyfile.global \
-  hello-c++-kde/admin/install-sh \
-  hello-c++-kde/admin/libtool.m4.in \
-  hello-c++-kde/admin/ltmain.sh \
-  hello-c++-kde/admin/Makefile.common \
-  hello-c++-kde/admin/missing \
-  hello-c++-kde/admin/mkinstalldirs \
-  hello-c++-kde/admin/ylwrap \
-  hello-c++-kde/po/Makevars \
-  hello-c++-kde/po/POTFILES.in \
-  \
   hello-c++-gnome/INSTALL \
   hello-c++-gnome/autogen.sh \
   hello-c++-gnome/autoclean.sh \
@@ -700,7 +663,6 @@ EXAMPLESDIRS = \
   hello-c++ \
   hello-c++20 \
   hello-c++-qt \
-  hello-c++-kde \
   hello-c++-gnome \
   hello-c++-wxwidgets \
   hello-objc \
