--- gettext-0.18.2/gettext-runtime/gnulib-lib/stdio.in.h.gets~	2012-12-25 06:29:09.000000000 +0100
+++ gettext-0.18.2/gettext-runtime/gnulib-lib/stdio.in.h	2012-12-26 19:22:14.146144798 +0100
@@ -707,10 +707,12 @@ _GL_WARN_ON_USE (getline, "getline is un
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
--- gettext-0.18.2/gettext-tools/gnulib-lib/stdio.in.h.gets~	2012-12-25 06:29:43.000000000 +0100
+++ gettext-0.18.2/gettext-tools/gnulib-lib/stdio.in.h	2012-12-26 19:22:44.985773371 +0100
@@ -707,10 +707,12 @@ _GL_WARN_ON_USE (getline, "getline is un
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
--- gettext-0.18.2/gettext-tools/libgettextpo/stdio.in.h.gets~	2012-12-25 06:30:20.000000000 +0100
+++ gettext-0.18.2/gettext-tools/libgettextpo/stdio.in.h	2012-12-26 19:23:21.877330837 +0100
@@ -707,10 +707,12 @@ _GL_WARN_ON_USE (getline, "getline is un
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
