diff -Naur gettext-0.19.5/gettext-runtime/gnulib-lib/stdio.in.h gettext-0.19.5.tpg/gettext-runtime/gnulib-lib/stdio.in.h
--- gettext-0.19.5/gettext-runtime/gnulib-lib/stdio.in.h	2015-07-10 06:38:08.000000000 +0000
+++ gettext-0.19.5.tpg/gettext-runtime/gnulib-lib/stdio.in.h	2015-07-15 22:40:39.557121439 +0000
@@ -722,10 +722,12 @@
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS && !defined __cplusplus
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
 struct obstack;
diff -Naur gettext-0.19.5/gettext-tools/gnulib-lib/stdio.in.h gettext-0.19.5.tpg/gettext-tools/gnulib-lib/stdio.in.h
--- gettext-0.19.5/gettext-tools/gnulib-lib/stdio.in.h	2015-07-10 06:38:39.000000000 +0000
+++ gettext-0.19.5.tpg/gettext-tools/gnulib-lib/stdio.in.h	2015-07-15 22:41:07.727880013 +0000
@@ -722,10 +722,12 @@
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS && !defined __cplusplus
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
 struct obstack;
diff -Naur gettext-0.19.5/gettext-tools/libgettextpo/stdio.in.h gettext-0.19.5.tpg/gettext-tools/libgettextpo/stdio.in.h
--- gettext-0.19.5/gettext-tools/libgettextpo/stdio.in.h	2015-07-10 06:39:18.000000000 +0000
+++ gettext-0.19.5.tpg/gettext-tools/libgettextpo/stdio.in.h	2015-07-15 22:41:33.800582094 +0000
@@ -722,10 +722,12 @@
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS && !defined __cplusplus
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
 struct obstack;
