--- gettext-0.18.2/gettext-tools/tests/Makefile.am.bero	2013-02-04 23:47:59.540749074 +0100
+++ gettext-0.18.2/gettext-tools/tests/Makefile.am	2013-02-04 23:52:16.559652946 +0100
@@ -187,8 +187,7 @@ TESTS_ENVIRONMENT = top_srcdir=$(top_src
 		    LOCALE_FR_UTF8='@LOCALE_FR_UTF8@' \
 		    LOCALE_JA='@LOCALE_JA@' \
 		    host_os='@host_os@' \
-		    CONFIG_SHELL='$(SHELL)' \
-		    $(SHELL)
+		    CONFIG_SHELL='$(SHELL)'
 
 # Update the expected test results.
 update-expected: xg-c-1.ok.po
